### PR TITLE
dcr: Add mix split tx identification.

### LINKED ISSUE #3057: dcr/ui: Mixing transacion still shown when purchasing tickets.
Trying out mixing and purchasing tickets on mainnet. Whenever I purchase a ticket with mixing turned on and `Hide Mixing Transactions` checked, I still see a mixing transaction.

image

I do not have 5k dcr to send:
https://explorer.dcrdata.org/tx/a1a5fffab8313114bb9b0cee6b768eb5f9648ff0cb042b0f96c9974a7d3a8d75

### PR DESCRIPTION
**Pull request by JoeGruffins**
closes  #3057

### CODE REVIEW

**Review by dev-warrior777 — APPROVED** [review]

**Comment by dev-warrior777** [comment]
"This dumbly checks for at least three outputs of the same size and three of other sizes."

How could it be less naive?

File: client/asset/dcr/dcr.go [hunk 1]
```diff
@@ -6126,7 +6157,7 @@ func (dcr *ExchangeWallet) idUnknownTx(ctx context.Context, ltxr *ListTransactio
 		}, nil
 	}
 
-	if isMix, mixDenom := isMixTx(tx.MsgTx); isMix {
+	if isMix, mixDenom := isMixTx(tx.MsgTx); isMix || isMixedSplitTx(tx.MsgTx) {
```
Comment (JoeGruffins) [inline client/asset/dcr/dcr.go:6160]: Oh, I guess this isn't a great place since we won't have the mixDenom...

Comment (JoeGruffins) [reply client/asset/dcr/dcr.go]: Moved to its own section https://github.com/decred/dcrdex/compare/867451f6f5eb71e76ed7537b1a73277e276a802f..909e4b5a918c4b1ad7797f7fd5cce6128ea4b6a6

File: client/asset/dcr/dcr.go [hunk 2]
```diff
@@ -6160,6 +6191,38 @@ func (dcr *ExchangeWallet) idUnknownTx(ctx context.Context, ltxr *ListTransactio
...
+			addr := addrs[0]
+			owns, err := dcr.wallet.WalletOwnsAddress(ctx, addr)
+			if err != nil {
+				dcr.log.Errorf("walletOwnsAddress error: %w", err)
+				continue
+			}
+
+			if owns {
+				mixedAmount += uint64(txOut.Value)
+			}
```
Comment (martonp) [inline client/asset/dcr/dcr.go:6215]: Wouldn't this also count the value of a change output? How about changing the `isMixedSplitTx` signature to also return the `ticketOutAmt` as it does in dcrdata. Then you could reuse the same block above where it checks `if txOut.Value == mixDenom { `.

Comment (JoeGruffins) [reply client/asset/dcr/dcr.go:6215]: Ok, I wasn't sure what we wanted to show. If just ticket amounts can do that.

Comment (JoeGruffins) [reply client/asset/dcr/dcr.go]: Only adding tickets for mixed amount https://github.com/decred/dcrdex/compare/909e4b5a918c4b1ad7797f7fd5cce6128ea4b6a6..74578b75da4ced4016b297655266295a75bdb374

Comment (martonp) [reply client/asset/dcr/dcr.go]: Looks like the inputs of these transactions are all the size of a mix output or a ticket redemption right? And the outputs are either the size of a ticket purchase or a change, is this correct? If so, then I think it makes sense to display the ticket amounts.

**Comment by martonp** [comment]
You could do this to avoid the duplication: https://github.com/decred/dcrdex/commit/29ee36d6696bdd11cf4d79757b397526bb54ade3

**Comment by JoeGruffins** [comment]
Ok added those changes!

**Review by martonp — APPROVED** [review]

**Pull request merged**